src/test/resources/checkstyle.xml

```diff
@@ -235,11 +235,6 @@ Change log:
         </module>
         <module name="NonEmptyAtclauseDescription"/>
         <module name="JavadocTagContinuationIndentation"/>
-        <module name="SummaryJavadoc">
```

Comment:
I find for many methods, this leads to redundant code. For basic functions, a javadoc
```
@returns an object transformed from input with blah blah behavior
```
seems totally sufficient. By requiring both a description and a returns flag, there is often comment duplication, resulting in methods I can point out in this repo like: 
```
        /**
         * Returns a new virtual shard iterator positioned at the sequence number of the first record in the list.
         *
         * @param records Record list. Must not be empty.
         * @return New shard iterator positioned at the first record in the list.
         */
        CachingShardIterator resolvePositionIterator(List<Record> records) {
```
```
        /**
         * Returns the DynamoDB shard iterator, may be empty.
         *
         * @return DynamoDB shard iterator
         */
        Optional<String> getDynamoDbIterator() {
```

```
        /**
         * Returns a result object for the given records that starts with a next iterator positioned after the last
         * record.
         *
         * @param records Record list. Must not be empty.
         * @return GetRecordsResult.
         */
```

In each of the above case, either the summary or the returns snippet can be condensed into a single snippet. I'm all about good documentation, but also, I believe in keeping it concise and to the point. More words in javadocs doesn't mean more value. 

```
 For functions doing interesting things, sure add both a description, and param/return snippets, but don't enforce it an all methods is my opinion.

Reply:
those examples offer something for everyone—those who prefer to look at the summary, those who prefer to look at `@param`/`@return`, and those who love 'em both!

if a developer feels that both are redundant, couldn't they leave out the `@return` part? I certainly do that from time to time.

Google (https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment) and Oracle (https://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#firstsentence) both note that that each Javadoc block begins with summary text—Google even explicitly notes that Javadoc that has only `@return` code should be replaced with a summary.